src/transforms/apply-operation.js

```diff
@@ -226,10 +326,10 @@ function removeText(state, operation) {
  */
 
 function setMark(state, operation) {
-  const { path, offset, length, mark, properties } = operation
+  const { path, offset, length, mark, newMark } = operation
```

Comment:
To have some context here, the lowest level method `Text.updateMark` was the one responsible of [updating the mark](https://github.com/ianstormtaylor/slate/blob/master/src/models/text.js#L290). It needed to have the existing `Mark` value, would then create a new `Mark` by merging properties with `Immutable.Map.merge`, and would replace the previous `Mark` value everywhere with the new value.
Since `Text.updateMark` creates the new `Mark`, we could not provide the new `Mark` yet when we needed to [create the inverse operation](https://github.com/ianstormtaylor/slate/blob/master/src/transforms/operations.js#L297). So the inverse operation was not updating anything.
I decided to move the update logic of the mark higher in the stack. It's more flexible, and allowed to fix the issue.

I think a `Mark` can be serialized to JSON easily. If not, then the existing `mark` argument was already a problem.